File: 0011 - Container With Most Water/solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l, r, res = 0, len(height)-1, 0
        while l < r:
            if height[l] < height[r]:
                res = max(res, (r-l)*height[l]); l += 1
            else:
                res = max(res, (r-l)*height[r]); r -= 1
        return res;

File: 0011 - Container With Most Water/solution.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l = 0, r = height.size()-1, res = 0;
        while(l < r){
            int h = min(height[l], height[r]);
            res = max(res, (r-l)*h);
            while(l<r && height[l]<=h) l++;
            while(l<r && height[r]<=h) r--;
        }
        return res;
    }
};